File: basic_prob/triplet_sum.cpp
#include<bits/stdc++.h>
using namespace std;

void triplet_sum(int a[], int n, int sum)
{
	sort(a, a+n);
	for(int i=0; i < n-2; i++)
	{
		int j = i+1, k = n-1;
		while(j < k)
		{
			if(a[i] + a[j] + a[k] >= sum)
				k--;
			else
			{
				for(int x = j+1; x<=k; x++)
					cout<<"("<<a[i]<<","<<a[j]<<","<<a[x]<<")"<<endl;
				j++;

			}

			
		}
	}
}


int main()
{
	int a[] = {5, 1, 3, 4, 7 };
	int n = sizeof(a)/sizeof(a[0]);
	int sum = 12;
	triplet_sum(a, n, sum);
	return 0;
}

File: basic_prob/triplet_sum.py
def triplet_sum(a, n, sum):
	a.sort()
	for i in range(0,n-2):
		j = i+1
		k = n-1
		while(j < k):

			if a[i] + a[j] +a[k] >= sum:
				k = k-1

			else:
				
				for x in range(j+1, k+1):
				    print(a[i],a[j],a[x])
				j = j+1


a = [5, 1, 3, 4, 7]
n = len(a)
sum = 12
triplet_sum(a, n, sum)
